src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs

```diff
@@ -541,12 +543,9 @@ private void WriteStartIndented(byte token)
                 output[BytesPending++] = JsonConstants.ListSeparator;
             }
 
-            if (_tokenType != JsonTokenType.PropertyName)
+            if ((_tokenType != JsonTokenType.PropertyName && _tokenType != JsonTokenType.None) || _commentAfterNoneOrPropertyName)
```

Comment:
Forget the exact syntax but could use pattern matching instead:
```suggestion
            if ((_tokenType is not JsonTokenType.PropertyName or is not JsonTokenType.None) || _commentAfterNoneOrPropertyName)
```

Reply:
Fixed.